### PR TITLE
Script to spin up a webserver for the docs

### PR DESCRIPTION
**Pull request by silky**
Specifically so we can view the haddocks (and make use of the search
capability).

Related to SCP-1900.

@gilligan Adding you as a reviewer so you can let me know if I've done this nix stuff in a nice way ... :) Very happy to change/receive some suggestions on how to do it better.

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-playground-client/default.nix [hunk 1]
```diff
@@ -42,6 +42,17 @@ let
     $(nix-build --quiet --no-build-output ../default.nix -A plutus-playground.server-invoker)/bin/plutus-playground webserver
   '';
 
+  # For dev usage
```
Comment (j-mueller) [inline plutus-playground-client/default.nix:45]: this should live with the other [convenience scripts](https://github.com/input-output-hk/plutus/blob/master/nix/pkgs/default.nix#L67-L83)

Comment (j-mueller) [reply plutus-playground-client/default.nix:45]: To be more precise, it should live under `plutus/nix/pkgs/serve-docs/default.nix` and be referenced from the other convenience scripts. At least that seems sensible to me.

Comment (silky) [reply plutus-playground-client/default.nix]: @j-mueller thanks for the suggestion; I've moved things around, hopefully it's nicer now :)

**Review by gilligan — CHANGES_REQUESTED** [review]

File: shell.nix [hunk 2]
```diff
@@ -83,6 +83,7 @@ let
     updateClientDeps
     updateMetadataSamples
     deployment.getCreds
+    docs.serve-docs
```
[the remark this answers is not in the document]
Comment (michaelpj) [reply shell.nix:86]: it's tiny, I can't imagine this matters

[the remark this answers is not in the document]
Comment (silky) [reply shell.nix:86]: no, no difference at all as far as i could tell

**Comment by silky** [comment]
Thanks @gilligan for your feedback; I've changed it all as you suggested :)

**Review by gilligan — APPROVED** [review]
LGTM

**Comment by michaelpj** [comment]
Ooops! @gilligan was actually totally right. I was looking at the `serve` function that just serves an arbitrary thing, `serve-docs` depends on the doc site which depends on literally all the Haskell packages, so that'll make the shell rebuild every time we touch anything. I think we almost certainly *don't* want that.

What I think would be safer would be to write a script that calls nix, i.e.
```bash
cd $(nix-build default.nix -A docs.site)
${python3}/bin/python3 -m http.server ${toString port}
```
Then the *script* doesn't depend on building `docs.site`.

**Comment by silky** [comment]
@michaelpj This sounds reasonable and makes total sense; I'll do that now.

**Pull request merged**